youtube_dl/extractor/safari.py

```diff
@@ -17,41 +14,38 @@
 
 
 class SafariBaseIE(InfoExtractor):
-    _LOGIN_URL = 'https://learning.oreilly.com/accounts/login/'
+    _LOGIN_URL = 'https://learning.oreilly.com/member/login/'
     _NETRC_MACHINE = 'safari'
 
     _API_BASE = 'https://learning.oreilly.com/api/v1'
     _API_FORMAT = 'json'
 
     LOGGED_IN = False
 
-    def _real_initialize(self):
-        self._login()
-
-    def _login(self):
-        username, password = self._get_login_info()
-        if username is None:
-            return
-
+    def _perform_login(self, username, password):
         _, urlh = self._download_webpage_handle(
-            'https://learning.oreilly.com/accounts/login-check/', None,
+            'https://learning.oreilly.com/member/login/', None,
             'Downloading login page')
 
         def is_logged(urlh):
-            return 'learning.oreilly.com/home/' in urlh.geturl()
+            url = urlh.geturl()
+            parsed_url = compat_urlparse.urlparse(url)
+            return parsed_url.hostname.endswith('learning.oreilly.com') and (
+                parsed_url.path.startswith('/home/')
+                or (parsed_url.path == '/member/login/' and not parsed_url.query))
 
         if is_logged(urlh):
```

Comment:
I think this should take a URL, not a handle, and `geturl()` should be used. Also, we can match www.learning.oreilly.com, learning.oreilly.com but not fakelearning.oreilly.com:
```suggestion
        def is_logged(url):
            parsed_url = compat_urlparse.urlparse(url)
            return parsed_url.hostname.split('.')[-3:] == ['learning', 'oreilly', 'com'] and (
                parsed_url.path.startswith('/home/')
                or (parsed_url.path == '/member/login/' and not parsed_url.query))

        redirect_url = urlh.geturl()

        if is_logged(redirect_url):
```